federation/pkg/dnsprovider/providers/aws/route53/internal/route53api.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/* internal implements a stub for the AWS Route53 API, used primarily for unit testing purposes */
+package internal
```

Comment:
How about putting it in testing package? And maybe name this fake_route53?
Example of other place where we do that: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/pod/testing/fake_mirror_client.go
(I dont have strong opinion on struct name, but I think naming the package testing is better than internal).